### ISSUE TITLE
New Payload - LEAD(), LAG() and CONCAT

### ISSUE THREAD

**Issue by burninatorsec** [sqlmapproject/sqlmap#4863]
**Is your feature request related to a problem? Please describe.**
I love SQLMap and use it all the time... however, while conducting an appsec assessment, I had a situation where SQLMap detected inline, error-based, and time-based bline SQL injections but was unable to exploit it via the SQL shell option. It was retrieving column and table names, but I really needed the row data. So I wrote a custom payload  using LEAD/LAG/CONCAT to achieve this using the error-based SQLi. I was unable to find an existing payload like it in SQLMap's source code. It would be great to see this integrated as part of SQLMap. Also, I used to be a developer before working in appsec, so I would be happy to do the pull request myself if needed.

**Describe the solution you'd like**
I have outlined a description of the solution I used here at my blog: https://burninatorsec.blogspot.com/2021/06/sql-injection-manual-data-exfiltration.html 

**Describe alternatives you've considered**
This appears to be a situation where the application was disallowing injections that use "WHERE" or "TOP", etc. , so perhaps any alternative keyword that can iterate over rows without those keywords would work. I just happened to end up using LEAD/LAG.

**Additional context**
As far as context, I found this payload useful because I was able to extract PII and other sensitive data that helped me privilege escalate. Hopefully it would be useful to someone else as well.

**Comment by burninatorsec** [comment]
> So, you have the other end blocking the `WHERE` and/or `TOP`, while expecting that it won't block the rest of SQL keywords. Either if I implement this payload(s), there will be usage of those keywords.

Yes, WHERE and TOP were blocked by the server. However, the payload I used with LEAD and LAG avoided WHERE and TOP completely. This is how the bypass was able to work without needing to use WHERE and TOP. 

Perhaps I'm misunderstanding, but are you saying it would be impossible to add the LEAD/LAG payload without WHERE or TOP? Because I was able to do this manually. I can provide more details than what I've written in the blog post if that would help.

**Comment by burninatorsec** [comment]
Based on what you've said, it seems that the limitation is that the SQL shell needs to be able to handle "where" queries, but my payload doesn't allow that. So, when I did this without SQLMap, I had to write a custom script. It would iterate over all the data rows using LEAD/LAG and use a separate script to check if there was a match in what the user was requesting in the "where" condition. Yes, it would be tedious to do manually, but that's exactly why it would be so great to automate as part of SQLMap if possible. Maybe this a rare scenario, but given how much work it took to implement, I'd be happy if it saves even one other person that work.
